src/Features/Core/Portable/Completion/Providers/AbstractKeywordCompletionProvider.cs

```diff
@@ -60,10 +60,17 @@ private async Task<ImmutableArray<CompletionItem>> RecommendCompletionItemsAsync
             if (syntaxFacts.IsInNonUserCode(syntaxTree, position, cancellationToken))
                 return ImmutableArray<RecommendedKeyword>.Empty;
 
-            using var _ = ArrayBuilder<RecommendedKeyword>.GetInstance(out var result);
+            using var _1 = ArrayBuilder<Task<ImmutableArray<RecommendedKeyword>>>.GetInstance(out var tasks);
             foreach (var recommender in _keywordRecommenders)
+                tasks.Add(Task.Run(() => recommender.RecommendKeywords(position, context, cancellationToken), cancellationToken));
```

Comment:
I am a little bit worried about here since it might start more than 100 threads.

Reply:
That's not what Task.Run does. It creates tasks for the thread pool itself to run. The thread pool then uses work stealing to attempt to keep an thread pool threads occupied with real work to do. 

The thing that makes many threads launch is when the thread pool threads themselves are blocked and the thread pool isn't making progress getting through tasks. When this happens, the thread pool starts spawning new threads to help out. That's a very very bad situation. But it doesn't happen when you have lots of tasks, it happens when you have misbehaving tasks. 

The thread pool itself is fine chewing through millions to billions of tasks :-)